### PR TITLE
feat(Partner Usage Reports): add new service to project

### PR DESCRIPTION
**Pull request by manukm-ibm**
## PR summary

Adding new partner API to fetch resource usage reports.
TRI: https://github.ibm.com/BSS/architecture/blob/master/design-docs/cosell/partner.md
Issue: https://github.ibm.com/BSS/metering-deliverables/issues/2063

## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## Does this PR introduce a breaking change?    
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by manukm-ibm** [comment]
Integration test results
Screenshot 2024-01-23 at 3 45 34 PM

Example test results
Screenshot 2024-01-23 at 3 47 58 PM

File: README.md [hunk 1]
```diff
@@ -60,6 +60,7 @@ Service Name | Artifact Coordinates
 [IAM Policy Management](https://cloud.ibm.com/apidocs/iam-policy-management?code=java) | com.ibm.cloud:iam-policy-management:0.48.2
 [IBM Cloud Shell](https://cloud.ibm.com/apidocs/cloudshell?code=java) | com.ibm.cloud:ibm-cloud-shell:0.48.2
 [Open Service Broker](https://cloud.ibm.com/apidocs/resource-controller/ibm-cloud-osb-api?code=java) | com.ibm.cloud:open-service-broker:0.48.2
+[Partner Usage Reports](https://cloud.ibm.com/apidocs/partner-apis/partner-usage-reports?code=java) | com.ibm.cloud:partner-usage-reports:0.48.2
```
Comment (padamstx) [inline README.md:63]: This link to the online reference docs is different than the one in the Go SDK.  These should probably be the same.

**Review by padamstx — COMMENTED** [review]
LGTM, except that the apidocs link is different than the link in the Go SDK.

**Review by padamstx — APPROVED** [review]
LGTM

**Pull request merged**